app.js

```diff
@@ -1763,6 +1837,80 @@ define(function(require) {
 			}
 		},
 
+		updateMFAConfiguration: function(MFAData, accountId, isOn) {
+			var self = this,
+				toastrMessages = self.i18n.active().multiFactorAuthentication.toastr;
+
+			monster.waterfall([
+				function(callback) {
+					self.accountsGetPhoneNumbers({
+						accountId: accountId
+					}, function(phoneNumbers) {
+						var allNumbers = _.get(phoneNumbers, 'numbers'),
+							numbersWithMessaging = [];
+
+						_.each(allNumbers, function(number, key) {
+							var hasMessaging = false;
+
+							_.each(number.features, function(feature) {
+								if (['sms', 'mms'].indexOf(feature) > -1) {
+									hasMessaging = true;
+								}
+							});
+
+							if (hasMessaging) {
+								numbersWithMessaging.push(key);
+							}
+						});
+						callback(null, numbersWithMessaging);
+					});
+				},
+				function(numbersWithMessaging, callback) {
+					var oomaBoxList = [];
+
+					self.accountsGetOomaSMSBoxes({
+						accountId: accountId
+					}, function(oomaSmsBoxes) {
+						_.each(oomaSmsBoxes, function(oomaSmsBox) {
+							_.each(oomaSmsBox.numbers, function(number) {
+								if (numbersWithMessaging.indexOf(number) > -1) {
+									oomaBoxList.push(number);
+								}
+							});
+						});
```

Comment:
This could be shortened a bit with Lodash:

```suggestion
						var oomaBoxList = _.chain(oomaSmsBoxes)
							.flatMap('numbers')
							.intersect(numbersWithMessaging)
							.value();
```